### PR TITLE
🧪 [testing] add tests for duplicate invite errors in POST /api/papers/:id/invites

### PR DESCRIPTION
**Pull request by is0692vs**
🎯 **What:** The testing gap addressed was missing tests for the database insertion error cases in the `POST /api/papers/:id/invites` route when sending coauthor invites.
📊 **Coverage:** Specifically, tests were added to ensure that a UNIQUE constraint failure on the `coauthor_invites` table correctly returns a 409 "Invite already sent", and that other database errors properly throw and result in a 500 response.
✨ **Result:** Test coverage improved for edge cases and database exceptions during invite operations.

---
*PR created automatically by Jules for task [4906311614618204708](https://jules.google.com/task/4906311614618204708) started by @is0692vs*

<details><summary><h3>Greptile Summary</h3></summary>

このPRは `POST /api/papers/:id/invites` エンドポイントに対して、データベース挿入時のエラーハンドリングをカバーする2つのテストケースを追加します。

- **409テスト**: `coauthor_invites` テーブルのUNIQUE制約違反（`\"UNIQUE constraint failed: ...\"` という形式のエラーメッセージ）をシミュレートし、`\"Invite already sent\"` という409レスポンスが返されることを検証
- **500テスト**: UNIQUE制約以外の汎用的なDBエラーが再スローされ、Honoのデフォルトエラーハンドラーによって500レスポンスとなることを検証
- `mockDb.select` の3回のモック（`isUploader` チェック → `alreadyAuthor` チェック → `invitee` 存在チェック）は、`papers.ts` ルート実装の実際の呼び出し順序と正確に一致している
- `authMiddleware` はJWT検証のみを行いDB参照しないため、追加のモックは不要であることも確認済み
- 軽微な点として、2つ目のテスト名がHTTPレスポンス（500）ではなく内部動作（`throw`）を説明しており、既存テストの命名規則と若干ずれがある
</details>

<h3>Confidence Score: 5/5</h3>

テストのみの変更でありプロダクションコードへの影響はなく、マージ可能です。

変更はテストファイルのみで、2つの新規テストケースはモック設定・実装との一致・アサーション内容いずれも正確。残る指摘はテスト名の命名スタイル（P2）のみ。

特に注意が必要なファイルはありません。

<details><summary><h3>Important Files Changed</h3></summary>

| Filename | Overview |
|----------|----------|
| apps/api/src/routes/__tests__/papers.test.ts | POST /api/papers/:id/invites の重複招待エラー（409）と予期しないDBエラー（500）に対する2つのテストケースを追加。モックの設定順序は実際のルート実装と一致しており、テストロジックは正確。 |

</details>

</details>

<h3>Sequence Diagram</h3>

```mermaid
sequenceDiagram
    participant Client
    participant Hono as Hono Router
    participant Auth as authMiddleware
    participant Route as POST /api/papers/:id/invites
    participant DB as mockDb (D1)

    Client->>Hono: POST /api/papers/paper-1/invites
    Hono->>Auth: verify JWT
    Auth-->>Hono: user = { sub: "user-uploader" }
    Hono->>Route: handle request

    Route->>DB: select (isUploader check)
    DB-->>Route: { role: "uploader" } ✓

    Route->>DB: select (alreadyAuthor check)
    DB-->>Route: null ✓

    Route->>DB: select (invitee exists check)
    DB-->>Route: { id: "user-invitee" } ✓

    alt UNIQUE constraint error
        Route->>DB: insert(coauthorInvites).values(...)
        DB-->>Route: Error("UNIQUE constraint failed: ...")
        Route-->>Client: 409 { error: "Invite already sent" }
    else Other DB error
        Route->>DB: insert(coauthorInvites).values(...)
        DB-->>Route: Error("Some other DB Error")
        Route->>Hono: throw e (re-throw)
        Hono-->>Client: 500 Internal Server Error
    end
```

<details><summary>Prompt To Fix All With AI</summary>

`````markdown
This is a comment left during a code review.
Path: apps/api/src/routes/__tests__/papers.test.ts
Line: 1012

Comment:
**テスト名が内部動作を説明しており、期待するHTTP応答が不明確**

`"throws non-UNIQUE errors"` というテスト名は、ルートハンドラーの内部動作（`throw e` による再スロー）を説明しています。しかし、テストが実際に検証しているのはHTTPレスポンスのステータスコード（500）です。他のテストの命名規則（例: `"returns 409 when invite already exists"`）と一致させると、より一貫性があります。

```suggestion
    it("POST /api/papers/:id/invites returns 500 for non-UNIQUE database errors", async () => {
```

How can I resolve this? If you propose a fix, please make it concise.
`````

</details>

<sub>Reviews (1): Last reviewed commit: ["🧪 \[testing\] add missing tests for dupli..."](https://github.com/hiroki-org/openshelf/commit/6c71d910fdf1e70c5e43a666d4428bd0a7a167f9) | [Re-trigger Greptile](https://app.greptile.com/api/retrigger?id=26668515)</sub>

> Greptile also left **1 inline comment** on this PR.

### CODE REVIEW

File: apps/api/src/routes/__tests__/papers.test.ts [hunk 1]
```diff
@@ -975,4 +975,68 @@ describe("papers routes", () => {
...
+            },
+            env as any
+        );
+
+        expect(res.status).toBe(409);
+        const data = await res.json();
+        expect(data.error).toBe("Invite already sent");
+    });
+
+    it("POST /api/papers/:id/invites throws non-UNIQUE errors", async () => {
```
[the remark this answers is not in the document]
Comment (is0692vs) [reply apps/api/src/routes/__tests__/papers.test.ts]: 対応しました: テスト名を `POST /api/papers/:id/invites returns 500 for non-UNIQUE database errors` に変更し、検証対象（HTTP 500）を明示しました。あわせて重複セットアップも整理済みです。検証: `npm run test -- apps/api/src/routes/__tests__/papers.test.ts` で成功しています。
